apps/event-system/services/api/clients.service.ts

```diff
@@ -164,19 +164,74 @@ module.exports = {
       },
     },
 
-    getByCustomerId: {
+    updateOnInvoicePaymentSuccess: {
       params: {
         customerId: 'string',
+        endDate: 'number',
       },
+
       async handler(ctx: any) {
         try {
           const client = await this.adapter.findOne({
             'billing.customerId': ctx.params.customerId,
           });
 
-          return client;
+          const updateDoc = await this.adapter.updateById(
+            client._id,
+            {
+              $set: {
+                'billing.subscription.valid': true,
+                'billing.subscription.reason': null,
+                'billing.subscription.endDate': ctx.params.endDate,
+              },
+            },
+            (doc: any) => {
+              return doc;
+            }
+          );
+
+          if (!updateDoc) {
+            return await ctx.call('error.404');
+          }
+
+          return updateDoc;
         } catch (error) {
+          return await ctx.call('error.404');
+        }
+      },
+    },
+
+    updateOnInvoicePaymentFailed: {
+      params: {
+        customerId: 'string',
+      },
+
+      async handler(ctx: any) {
+        try {
+          const client = await this.adapter.findOne({
+            'billing.customerId': ctx.params.customerId,
+          });
+
+          const updateDoc = await this.adapter.updateById(
+            client._id,
+            {
+              $set: {
+                'billing.subscription.valid': false,
+                'billing.subscription.reason': 'payment_failed',

```

Comment:
You can set the reason like this: `payment-failed` to stay consistent with our data system